src/models/searchservice/response/HighlightedFields.ts

```diff
@@ -0,0 +1,41 @@
+import { HighlightedValue } from './HighlightedValue';
+
+/**
+ * A mapping of the values emphasized by the Answers API.
+ *
+ * @example
+ * If a user searches for 'pet' and a description field in the results contains the value 'A pet store',
+ * the API will likely match the word 'pet'.
+ *
+ * @example
+ * ```js
+ * {
+ *   name: {
+ *     value: 'Yext',
+ *     matchedSubstrings: [{ offset: 0, length: 4}]
+ *   },
+ *   c_food: [
+ *     {
+ *       fruits: [
+ *       {
+ *         value: 'apple',
+ *         matchedSubstrings: []
+ *       },
+ *       {
+ *         value: 'pear',
+ *         matchedSubstrings: []
+ *       }
+ *     }
+ *   ]
+ * }
+ * ```
+ *
+ * @public
+ */
+export interface HighlightedFields {
+  /**
+   * A mapping of a field to either an array of HighlightedFields, HighlightedFields for
+   * this field's subfields, or a {@link HighlightedValue} for the field.
+   */
+  [fieldId: string]: HighlightedValue | HighlightedValue[] | HighlightedFields | HighlightedFields[]
```

Comment:
It doesn't look like the api-documenter supports this data shape in the generated docs, so I think we want to explain the data type at the top under an `@remarks` tag. I would put something along the lines of: "HighlightedFields is an object of arbitrary shape which contains `{@link HighlightedValue}` objects.

Reply:
updated!

Reply:
It looks like the docs generation does actually support this if we define it as a type rather than an interface. Can you try that? I think it would look like `type HighlightedFields = { [fieldId: string] : HighlightedValue | HighlightedValue[] | HighlightedFields | HighlightedFields[] }`

Reply:
let me try

Reply:
works!

Reply:
Now it looks like the description and remarks within the type doesn't show up in the docs. I'm not sure that we need the section "A mapping of a field to either..." anymore since the shape is now described in the docs. The remark "HighlightedFields is an object of arbitrary shape..." may fit at the top of the current remarks. Thoughts?